### ISSUE TITLE
problems with tree view?

### ISSUE THREAD

**Issue by GoogleCodeExporter** [eadevlin/autokey#111]
```
I'm using AutoKey (GTK UI) 0.71.3 under LMDE and I have not tested the qt 
version.

Using only the default scripts and phrases, a number of repeatable problems are 
evident:

- dragging between folders does not work
- dragging a script to the Addresses folder causes the My Phrases folder to 
disappear
- exporting and then importing the My Phrases folder fails with "maximum 
recursion depth exceeded"
```

Original issue reported on code.google.com by `mon...@gmail.com` on 4 Apr 2011 at 8:20

**Comment by GoogleCodeExporter** [comment]
```
Thanks for reporting. I'm surprised nobody else has noticed this till now! It 
wouldn't affect the QT version.
```

Original comment by `cdekter` on 4 Apr 2011 at 9:17
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Surprising to me too! Thanks for writing this useful tool!

I guess the recursion limit may be working as expected but as one must import 
into an existing folder, it's difficult avoid, especially if one wants to 
export the entire structure to a single file.
```

Original comment by `mon...@gmail.com` on 4 Apr 2011 at 9:47

**Comment by GoogleCodeExporter** [comment]
```
I'm afraid most of these problems are caused by a defect that seems to have 
been introduced in a recent version of GTK. The drag data is not correctly 
reporting the path of the source row(s) for drag/drop operations. There is 
nothing I can do about it.

The importing problem I couldn't reproduce, if you have time please provide 
more detailed steps to reproduce.
```

Original comment by `cdekter` on 5 Apr 2011 at 12:52

**Comment by GoogleCodeExporter** [comment]
```
Do you have a link to that gtk bug?

1. export the My Phrases folder
2. import the file which was just created

As it appears to me, folders can only be nested one level. I can successfully 
export and import the Sample Scripts folder (which has no nested folders).

If I delete all folders, import and folder/item creation are disabled.
```

Original comment by `mon...@gmail.com` on 5 Apr 2011 at 2:40

**Comment by GoogleCodeExporter** [comment]
```
I searched but didn't find it in the gnome tracker... basically when you 
receive the drag data event, you receive the selection object which contains a 
list of currently selected rows/columns. This has always worked fine, yet now 
suddenly it always reports the first row as being selected, instead of the 
actually selected row.

Thanks for the extra info, will try it and see.
```

Original comment by `cdekter` on 5 Apr 2011 at 3:18

**Comment by GoogleCodeExporter** [comment]
```
Fixed in version v0.80.0
```

Original comment by `cdekter` on 11 Sep 2011 at 11:30
- Changed state: **Fixed**
